Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -18,78 +18,108 @@ This file contains basics about the separable degree of a field extension.
 
-## TODO
+- `Polynomial.natSepDegree_eq_of_splits`: if a polynomial splits over `E`, then its separable degree
+  is equal to the number of distinct roots of it over `E`.
 
-- If `E / F` is a finite extension, then $[E:F]_s \mid [E:F]$. Thus we can define the inseparable
-  degree $[E:F]_i$ to be $[E:F]$ divided by $[E:F]_s$.
+- `Polynomial.natSepDegree_eq_of_isAlgClosed`: the separable degree of a polynomial is equal to
+  the number of distinct roots of it over any algebraically closed field.
 
-- Define the separable degree of a polynomial over a field to be the number of distinct roots
-  in algebraic closure, and prove that it's equal to `Polynomial.HasSeparableContraction.degree`.
+- `Polynomial.natSepDegree_expand`: if a field `F` is of exponential characteristic
+  `q`, then `Polynomial.expand F (q ^ n) f` and `f` have the same separable degree.
 
-- If `a` is an element algebraic over `F`, then $[F(a):F]_s$ is equal to the separable degree of
-  the minimal polynomial of `a` over `F`. In particular, `a` is separable over `F` if and only if
-  $[F(a):F]_s = [F(a):F]$.
+- `Polynomial.HasSeparableContraction.natSepDegree_eq`: if a polynomial has separable
+  contraction, then its separable degree is equal to its separable contraction degree.
 
-- If `E / F` is a finite extension, then $[E:F]_s = [E:F]$ if and only if every element of `E` is
-  separable over `F`, namely, `E / F` is a separable extension.
+- `Irreducible.natSepDegree_dvd_natDegree`: the separable degree of an irreducible
+  polynomial divides its degree.
 
-- As a corollary, if `S` is a subset of `E` consisting of separable elements, then `F(S) / F` is a
-  separable extension.
+- `IntermediateField.finSepDegree_adjoin_simple_eq_natSepDegree`: the (finite) separable degree of
+  `F⟮α⟯ / F` is equal to the separable degree of the minimal polynomial of `α` over `F`.
 
-- Define maximal separable subextension (or called relative separable closure)
-  `separableClosure F E : IntermediateField F E` of `E / F`, and prove that its degree over `F`
-  is $[E:F]_s$ if `E / F` is finite.
+- `IntermediateField.finSepDegree_adjoin_simple_eq_finrank_iff`: if `α` is algebraic over `F`, then
+  the separable degree of `F⟮α⟯ / F` is equal to the degree of `F⟮α⟯ / F` if and only if `α` is a
+  separable element.
 
-- Prove that `separableClosure F (AlgebraicClosure F)` is a separable closure of `F`.
+- `Field.finSepDegree_dvd_finrank`: the separable degree of any field extension `E / F` divides
+  the degree of `E / F`.
 
-- Prove that `[E:F]_s = 1` if and only if `E / F` is purely inseparable.
+- `Field.finSepDegree_le_finrank`: the separable degree of a finite extension `E / F` is smaller
+  than the degree of `E / F`.
+
+- `Field.finSepDegree_eq_finrank_iff`: if `E / F` is a finite extension, then its separable degree
+  is equal to its degree if and only if it is a separable extension.
+
+- `IntermediateField.isSeparable_adjoin_simple_iff_separable`: `F⟮x⟯ / F` is a separable extension
```

Comment:
Side comment: I think the phrase `adjoin_simple` is strange. It's a **simple** extension obtained by **adjoining** a **singleton**. Indeed [similar constructions](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Adjoin/Basic.html#Algebra.self_mem_adjoin_singleton) are called `adjoin_singleton`, but I think we could just use the shorthand `simple` here which also agrees with math terminology. I think we might open a PR to rename all `adjoin_simple` to `simple`.

Reply:
But `mem_adjoin_simple_self` -> `mem_simple_self` sounds strange ...